dev/benchmarks/platform_channels_benchmarks/lib/main.dart

```diff
@@ -0,0 +1,177 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+import 'dart:typed_data';
+
+import 'package:flutter/foundation.dart' show kDebugMode;
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import 'package:microbenchmarks/common.dart';
+
+const int _numMessages = 2500;
+
+List<Object> _makeTestBuffer(int size) {
+  final List<Object> answer = <Object>[];
+  for (int i = 0; i < size; ++i) {
+    switch (i % 9) {
+      case 0:
+        answer.add(1);
+        break;
+      case 1:
+        answer.add(math.pow(2, 65));
+        break;
+      case 2:
+        answer.add(1234.0);
+        break;
+      case 3:
+        answer.add(null);
+        break;
+      case 4:
+        answer.add(<int>[1234]);
+        break;
+      case 5:
+        answer.add(<String, int>{'hello': 1234});
+        break;
+      case 6:
+        answer.add('this is a test');
+        break;
+      case 7:
+        answer.add(true);
+        break;
+      case 8:
+        answer.add(Uint8List(64));
+        break;
+    }
+  }
+  return answer;
+}
+
+Future<double> _runBasicStandardSmall(
+    BasicMessageChannel<Object> basicStandard) async {
+  final Stopwatch watch = Stopwatch();
+  watch.start();
+  for (int i = 0; i < _numMessages; ++i) {
+    await basicStandard.send(1234);
+  }
+  watch.stop();
+  return watch.elapsedMicroseconds / _numMessages;
+}
+
+Future<double> _runBasicStandardLarge(
+    BasicMessageChannel<Object> basicStandard, List<Object> largeBuffer) async {
+  int size = 0;
+  final Stopwatch watch = Stopwatch();
+  watch.start();
+  for (int i = 0; i < _numMessages; ++i) {
+    final List<Object> result =
+        await basicStandard.send(largeBuffer) as List<Object>;
+    // This check should be tiny compared to the actual channel send/receive.
+    size += (result == null) ? 0 : result.length;
+  }
+  watch.stop();
+
+  if (size != largeBuffer.length * _numMessages) {
+    throw Exception(
+        'There is an error with the echo channel, the results don\'t add up: $size');
```

Comment:
nit: no new line here, or add a comma to the end of the string and put the closing paren on a new line.

Reply:
I tried that and it didn't look any better to my eyes.  Having a trailing comma for a parameter list is odd, it's not like an array or map literal.  This is what the automatic formatter wants in this case.

Reply:
It's what the style guide wants: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#if-you-have-a-newline-after-some-opening-punctuation-match-it-on-the-closing-punctuation

https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#prefer-a-maximum-line-length-of-80-characters

We don't use dartfmt in this repo, although it can come close if you specify a longer line length like 120 or something.

Reply:
Done, imo `If you have a newline after some opening punctuation, match it on the closing punctuation.` doesn't make any sense.  What is "it" and what should it be matched with?  I read it about 20 times trying to understand how it applies to this case and I still don't get it but from the examples it seems applicable =P